src/main/decompInitMod.F90

```diff
@@ -574,25 +630,52 @@ subroutine decompInit_clumps(lni,lnj,glc_behavior)
     enddo
 
     do n = 1,nclumps
+       ! Only do the error checking over the local processor
+       if (clumps(n)%owner == iam) then
        if (clumps(n)%ncells   /= allvecg(n,1) .or. &
            clumps(n)%nlunits  /= allvecg(n,2) .or. &
            clumps(n)%ncols    /= allvecg(n,3) .or. &
            clumps(n)%npatches /= allvecg(n,4) .or. &
            clumps(n)%nCohorts /= allvecg(n,5)) then
 
-          write(iulog ,*) 'decompInit_glcp(): allvecg error ncells ',iam,n,clumps(n)%ncells   ,allvecg(n,1)
-          write(iulog ,*) 'decompInit_glcp(): allvecg error lunits ',iam,n,clumps(n)%nlunits  ,allvecg(n,2)
-          write(iulog ,*) 'decompInit_glcp(): allvecg error ncols  ',iam,n,clumps(n)%ncols    ,allvecg(n,3)
-          write(iulog ,*) 'decompInit_glcp(): allvecg error patches',iam,n,clumps(n)%npatches ,allvecg(n,4)
-          write(iulog ,*) 'decompInit_glcp(): allvecg error cohorts',iam,n,clumps(n)%nCohorts ,allvecg(n,5)
+          write(iulog ,*) 'allvecg error: iam,n ',iam,n
+          write(iulog ,*) 'allvecg error ncells,allvecg ',iam,n,clumps(n)%ncells   ,allvecg(n,1)
+          write(iulog ,*) 'allvecg error lunits,allvecg ',iam,n,clumps(n)%nlunits  ,allvecg(n,2)
+          write(iulog ,*) 'allvecg error ncols,allvecg  ',iam,n,clumps(n)%ncols    ,allvecg(n,3)
+          write(iulog ,*) 'allvecg error patches,allvecg',iam,n,clumps(n)%npatches ,allvecg(n,4)
+          write(iulog ,*) 'allvecg error cohorts,allvecg',iam,n,clumps(n)%nCohorts ,allvecg(n,5)
 
-          call endrun(msg=errMsg(sourcefile, __LINE__))
+          call endrun(msg="allvecg error cohorts", file=sourcefile, line=__LINE__)
+          return
+       endif
        endif
     enddo
 
     deallocate(allvecg,allvecl)
     deallocate(lcid)
 
+    ! ------ Reset the clump type array for all non-local cid's to -1 to show it can be made smaller
```

Comment:
```suggestion
    ! ------ Reset the clump type array for all non-local cid's to -1 to show it can be made smaller
    ! TODO: Remove this when https://github.com/ESCOMP/CTSM/issues/3466 is done
```